motionplan/rrt.go

```diff
@@ -270,16 +258,28 @@ func sumCosts(path []node) float64 {
 	return cost
 }
 
-func transformNodes(path []node, transformBy spatialmath.Pose) []node {
-	transformedNodes := []node{}
-	for _, n := range path {
-		newNode := &basicNode{
-			q:      n.Q(),
-			cost:   n.Cost(),
-			pose:   spatialmath.Compose(n.Pose(), transformBy),
-			corner: n.Corner(),
+type rrtPlan struct {
+	*SimplePlan
```

Comment:
1. Do we ever want `SimplePlan` to be nil?
2. If not, why not embed the `SimplePlan` rather than a pointer to it?

Reply:
ok changing to a normal struct.